File: problems/371_sum_of_two_integers/solution.py
class Solution:
    def getSum(self, a: int, b: int) -> int:
        MASK = 0xFFFFFFFF
        MAX = 0x7FFFFFFF
        while b & MASK:
            carry = (a & b) << 1
            a = a ^ b
            b = carry
        return a & MASK if a > MAX else a & MASK if (a & MASK) <= MAX else ~(a ^ MASK)

File: problems/371_sum_of_two_integers/solution.cpp
#include <vector>
#include <string>
#include <unordered_map>
#include <unordered_set>
#include <algorithm>
#include <climits>
#include <queue>
#include <stack>
#include <cmath>
#include <numeric>
#include <set>
#include <map>
using namespace std;

class Solution {
public:
    int getSum(int a, int b) {
        while (b != 0) {
            int carry = a & b;
            a = a ^ b;
            b = (unsigned)carry << 1;
        }
        return a;
    }
};
